Source/SwiftLintFramework/Rules/CyclomaticComplexityRule.swift

```diff
@@ -51,25 +54,50 @@ public struct CyclomaticComplexityRule: ASTRule, ConfigProviderRule {
         return []
     }
 
-    private func measureComplexity(substructure: [SourceKitRepresentable]) -> Int {
-        return substructure.reduce(0) { complexity, subItem in
+    private func measureComplexity(file: File,
+                                   dictionary: [String: SourceKitRepresentable]) -> Int {
+        var hasSwitchStatements = false
+
+        let substructure = dictionary["key.substructure"] as? [SourceKitRepresentable] ?? []
+
+        let complexity = substructure.reduce(0) { complexity, subItem in
             guard let subDict = subItem as? [String: SourceKitRepresentable],
                       key = subDict["key.kind"] as? String else {
                 return complexity
             }
-            if let subSubItem = subDict["key.substructure"] as? [SourceKitRepresentable] {
-                return complexity +
-                    Int(complexityStatements.contains(key)) +
-                    measureComplexity(subSubItem)
+            if key == "source.lang.swift.stmt.switch" {
+                hasSwitchStatements = true
             }
-            return complexity + Int(complexityStatements.contains(key))
+
+            return complexity +
+                    Int(complexityStatements.contains(key)) +
+                    measureComplexity(file, dictionary: subDict)
         }
+
+        if hasSwitchStatements {
+            return reduceSwitchComplexity(complexity, file: file, dictionary: dictionary)
+        }
+
+        return complexity
+    }
+
+    // Switch complexity is reduced by `fallthrough` cases
+
+    private func reduceSwitchComplexity(complexity: Int, file: File,
+                                        dictionary: [String: SourceKitRepresentable]) -> Int {
+        let bodyOffset = Int(dictionary["key.bodyoffset"] as? Int64 ?? 0)
+        let bodyLength = Int(dictionary["key.bodylength"] as? Int64 ?? 0)
+
+        let c = (file.contents as NSString)
+            .substringWithByteRange(start: bodyOffset, length: bodyLength) ?? ""
+
+        let fallthroughCount = c.componentsSeparatedByString("fallthrough").count - 1
+        return complexity - fallthroughCount
     }
 
     private let complexityStatements = [
         "source.lang.swift.stmt.foreach",
```

Comment:
it'd be nice to get these into SourceKitten much like `SyntaxKind` and `SwiftDeclarationKind`.